### DISCUSSION TITLE
Why do the rows and columns seem to be reversed in to_sparse_tensor? Thanks a lot.

### DISCUSSION THREAD

**Discussion by storyandwine** [pyg-team/pytorch_geometric#4901] [Q&A] [answered]
In to_sparse_tensor  row = edge_index[1], col = edge_index[0]
https://github.com/pyg-team/pytorch_geometric/blob/e8b6defb2fb4f734cde6784a3761fdaf12161e8c/torch_geometric/transforms/to_sparse_tensor.py#L62-L66
While in other functions like dropout_adj
https://github.com/pyg-team/pytorch_geometric/blob/2db5effbbedec82d11159bb799978f98650421c6/torch_geometric/utils/dropout.py#L47
row = edge_index[0] ,col = edge_index[1]

**Comment by rusty1s** [comment, marked as answer]
Yes, this is due to `ToSparseTensor` saving the transposed adjacency matrix `adj_t` (which accelerates message passing). The transposed adjacency matrix is given by simply swapping `row` and `col`.
